main.tf

```diff
@@ -337,25 +361,64 @@ data "aws_iam_policy_document" "lb_log_delivery" {
       values   = ["bucket-owner-full-control"]
     }
   }
+}
 
-  statement {
-    sid = "AWSLogDeliveryAclCheck"
+# Cloudtrail
+data "aws_iam_policy_document" "cloudtrail_log_delivery" {
+  count = var.create_bucket && var.attach_cloudtrail_log_delivery_policy ? 1 : 0
 
-    effect = "Allow"
+  statement {
+    sid = "AWSCloudTrailAclCheck20150319"
 
     principals {
       type        = "Service"
-      identifiers = ["delivery.logs.amazonaws.com"]
+      identifiers = ["cloudtrail.amazonaws.com"]
     }
 
+    effect = "Allow"
+
     actions = [
       "s3:GetBucketAcl",
     ]
 
     resources = [
       aws_s3_bucket.this[0].arn,
     ]
+  }
+
+  statement {
+    sid = "AWSCloudTrailWrite20150319"
 
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+
+    actions = [
+      "s3:PutObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.this[0].arn}/${local.policy_cloudtrail_log_prefix}AWSLogs/${data.aws_caller_identity.this.account_id}/*",
```

Comment:
Could you please split this PR into two?
1. Add cloudtrail bucket policy with wider access (similar to how it is implemented now for other types of policies).
2. Tighten (as you call it, "security" improvement) existing policies via separate argument (e.g. `var.restrict_log_delivery_policies = true` (`false` by default to keep backward-compatibility))

Reply:
Ok I do that, I will propose another PR with modifications for the other paths.